Rewrite magic bitboard

lucasart opened the pull request:
Retain only the magic numbers from the original implementation (any magic numbers will do).

No functional change.

BENCH : 8,271,334

lucasart committed:
Rewrite magic bitboard

Retain only the magic numbers from the original implementation (any magic numbers will do).

No functional change.

BENCH : 8,271,334

lucasart committed:
Cleanup
- retire duplicate functions
- brace on same line for functions
- remove the unsigned cast trick

AndyGrant reviewed (commented):
Last 5 things I think. Will compile this again and merge once you change those.
Nice work!

lucasart committed:
Cosmetic fix

lucasart committed:
More cleanup

lucasart committed:
magics -> attacks

A better name for this unit, which is not about magics, but about attack bitboards
(for all pieces not just sliders). Magics is more an internal implementation detail.
